src/lua/httpc.c

```diff
@@ -155,15 +155,36 @@ luaT_httpc_request(lua_State *L)
 	if (req == NULL)
 		return luaT_error(L);
 
+	const char* content_type = NULL;
 	double timeout = TIMEOUT_INFINITY;
 	int max_header_name_length = MAX_HTTP_HEADER_NAME_LEN;
 	if (lua_isstring(L, 4)) {
 		size_t len = 0;
 		const char *body = lua_tolstring(L, 4, &len);
+		if (len > 0 && httpc_set_body(req, body, len) != 0) {
+			httpc_request_delete(req);
+			return luaT_error(L);
+                }
+        } else if ((lua_istable(L, 4)) &&
+                   (strcasecmp(method, "post") == 0)) {
```

Comment:
I would add an option to control this behaviour. Consider the following cases:

1. One may want to use json encoding for a string (quotes, escape sequences and so on).
2. The same for number and decimal.
3. One may want to pass arbitrary object (PoD/table or cdata/userdata with `__serialize`).

I don't know, what is better: assume JSON encoding for a table (without the option) or raise an error. A simple is better? An explicit is better?

Reply:
We agreed on some behavior that seems to be well-defined (see linked RFC), so my questions in this discussion look irrelevant now.